File: Java/Easy/Accepted/find-nearest-point-that-has-the-same-x-or-y-coordinate_7.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        
        int minDis = Integer.MAX_VALUE;
        int minDisIndex = -1;
        for(int i = 0; i < points.length; i++) {
            if(points[i][0] == x || points[i][1] == y) {
                int dis = Math.abs(x - points[i][0]) + Math.abs(y - points[i][1]);
                if(dis < minDis) {
                    minDis = dis;
                    minDisIndex = i;
                }
            }
        }
        return minDisIndex;
    }
}

File: Rust/Easy/Accepted/find-nearest-point-that-has-the-same-x-or-y-coordinate_7.rs
impl Solution {
    pub fn nearest_valid_point(x: i32, y: i32, points: Vec<Vec<i32>>) -> i32 {
        
        let mut min = std::i32::MAX;
        let mut idx = -1;
        for (i, p) in points.iter().enumerate() {
            let dx = (x - p[0]).abs();
            let dy = (y - p[1]).abs();
            if dx == 0 || dy == 0 {
                let d = dx + dy;
                if d < min {
                    min = d;
                    idx = i as i32;
                }
            }
        }
        idx
    }
}